tests/e2e_tests/kgain_e2e.py

```diff
@@ -0,0 +1,125 @@
+    print(l1_datadir)
+    l1_datadir_nonlin = os.path.join(tvacdata_path, "TV-20_EXCAM_noise_characterization", "nonlin")
+    nonlin_path = os.path.join(tvacdata_path, "TV-20_EXCAM_noise_characterization", "results", "nonlin_table_240322.txt")
+    kgain_result = os.path.join(tvacdata_path, "TV-20_EXCAM_noise_characterization", "results", "kgain_read_noise.txt")
+
+    # make output directory if needed
+    kgain_outputdir = os.path.join(e2eoutput_path, "l1_to_kgain_output")
+    if os.path.exists(kgain_outputdir):
+        shutil.rmtree(kgain_outputdir)
+    os.mkdir(kgain_outputdir)
+
+    # define the raw science data to process
+
+    l1_data_filelist_same_exp = [os.path.join(l1_datadir, "CGI_EXCAM_L1_00000{0}.fits".format(i)) for i in np.arange(51841,51851)]#[51841, 51870]] # just grab some files of it
+    print(l1_data_filelist_same_exp)
+    l1_data_filelist_range_exp = [os.path.join(l1_datadir, "CGI_EXCAM_L1_00000{0}.fits".format(i)) for i in np.arange(51731, 51761)]#[51731, 51840]]
+    print(l1_data_filelist_range_exp)
+    l1_data_filelist = [os.path.join(l1_datadir, "CGI_EXCAM_L1_00000{0}.fits".format(i)) for i in np.arange(51731, 51871)]
+    mock_cal_filelist = [os.path.join(l1_datadir_nonlin, "CGI_EXCAM_L1_00000{0}.fits".format(i)) for i in [51825, 55165]] # grab some real data to mock the calibration 
+    
+    ###### Setup necessary calibration files
+    # Create necessary calibration files
+    # we are going to make a new nonlinear calibration file using
+    # a combination of the II&T nonlinearity file and the mock headers from
+    # our unit test version of the NonLinearityCalibration
+    nonlin_dat = np.genfromtxt(nonlin_path, delimiter=",")
+    pri_hdr, ext_hdr = mocks.create_default_headers()
+    ext_hdr["DRPCTIME"] = time.Time.now().isot
+    ext_hdr['DRPVERSN'] =  corgidrp.__version__
+    mock_input_dataset = data.Dataset(mock_cal_filelist)
+    nonlinear_cal = data.NonLinearityCalibration(nonlin_dat,
+                                                 pri_hdr=pri_hdr,
+                                                 ext_hdr=ext_hdr,
+                                                 input_dataset=mock_input_dataset)
+    nonlinear_cal.save(filedir=kgain_outputdir, filename="mock_nonlinearcal.fits" )
+
+    # add calibration file to caldb
+    this_caldb = caldb.CalDB()
+    this_caldb.create_entry(nonlinear_cal)
+
+    ####### Run the walker on some test_data
+
+    for file in l1_data_filelist_same_exp:
+        image = data.Image(file)
+        # This should not be necessary anymore after the updates of the OBSTYPE keyword, up to now it is only "SCI"
+        if image.pri_hdr['OBSTYPE'] != 'MNFRAME':
+            image.pri_hdr['OBSTYPE'] = 'MNFRAME'
+            image.save(filename = file)
+        l1_data_filelist_range_exp.append(file)
+
+    walker.walk_corgidrp(l1_data_filelist_range_exp, "", kgain_outputdir, template="l1_to_kgain.json")
```

Comment:
Should we edit `walker.guess_template()` to handle how to process kgain without needing to pass into the template. Or can we not differentiate this data from the nonlinearity data?